File: src/1131-Maximum-of-Absolute-Value-Expression/1131.cpp
class Solution 
{
public:
    int maxAbsValExpr(vector<int>& arr1, vector<int>& arr2) 
    {
        int res = 0, n = arr1.size();
        int dire[4][2] = {{1, 1}, {-1, 1}, {1, -1}, {-1, -1}};
        for (auto& d : dire)
        {
            int r = arr1[0]*d[0] + arr2[0]*d[1];
            for (int i = 0; i < n; ++i)
            {
                int l = arr1[i]*d[0] + arr2[i]*d[1] + i;
                res = max(res, l - r);
                r = min(l, r);
            }
        }
        return res;
    }
};

File: src/1131-Maximum-of-Absolute-Value-Expression/1131.js
var maxAbsValExpr = function(arr1, arr2) {
    var res = 0, n = arr1.length;
    var dire = [[1, 1], [1, -1], [-1, 1], [-1, -1]];
    for (let op of dire) {
        var r = arr1[0] * op[0] + arr2[0] * op[1];
        for (let i = 0; i < n; ++i) {
            l = arr1[i] * op[0] + arr2[i] * op[1] + i;
            res = Math.max(res, l - r);
            r = Math.min(l, r);
        }
    }
    return res;
};